src/charm.py

```diff
@@ -247,6 +256,42 @@ def _hydra_layer(self) -> Layer:
                 },
             },
         }
+
+        layer_with_trace_config = {
```

Comment:
The only difference between `layer_config` and `layer_with_trace_config` are the environment variables. Can we add that part based on the `self._tracing_ready` check instead of keeping both layers separate?

Reply:
sure